src/web/index.php

```diff
@@ -1,28 +1,174 @@
  * (see Filtering tasks 1 and 2 below)
+
+ * @param string $filterLetter
+ * @param array $airports
+ * @return array
+ */
+function firstLetterFilter(string $filterLetter, array $airports): array
+{
+
+    $filteredArr = [];
+
+    foreach ($airports as $airport) {
+        if (mb_substr($airport['name'], 0, 1) === $filterLetter) {
+            $filteredArr[] = $airport;
+        }
+    }
+
+    return $filteredArr;
+
+}
+
+/**
+ * @param string $filterState
+ * @param array $airports
+ * @return array
  */
+function stateFilter(string $filterState, array $airports): array
+{
+
+    $filteredArr = [];
+
+    foreach ($airports as $airport) {
+        if ($airport['state'] === $filterState) {
+            $filteredArr[] = $airport;
+        }
+    }
+
+    return $filteredArr;
+
+}
+
+if ($filterLetter) {
+    $airports = firstLetterFilter($filterLetter, $airports);
+}
+
+if ($filterState) {
+    $airports = stateFilter($filterState, $airports);
+}
 
 // Sorting
 /**
  * Here you need to check $_GET request if it has sorting key
  * and apply sorting
  * (see Sorting task below)
+
+ * @param string $sortKey
+ * @param array $airports
+ * @return array
  */
+function codeSorting(string $sortKey, array $airports): array
+{
+
+    $unFilteredArr = $airports;
+    $sortKeyArr = array_column($airports, $sortKey);
+    $ind = array_multisort($sortKeyArr, SORT_ASC, $airports);
+
+    if ($ind) {
+        return $airports;
+    } else {
+        return $unFilteredArr;
+    }
+
+}
+
+if ($filterSort) {
+    $airports = codeSorting($filterSort, $airports);
+}
 
 // Pagination
 /**
  * Here you need to check $_GET request if it has pagination key
  * and apply pagination logic
  * (see Pagination task below)
+
+ * @param $filterPage
+ * @param array $airports
+ * @param array $filters
+ * @return void
+ */
+function pagination($filterPage, array $airports, array $filters): void {
+
+    $filterLetter = $_GET['filter_by_first_letter'] ?? null;
+    $filterState = $_GET['filter_by_state'] ?? null;
+    $filterSort = $_GET['sort'] ?? null;
+    $pageAmong = count($airports) / PAGINATING_INDEX + 1;
+    $href = '';
+
+    foreach ($filters as $filter) {
+        if ($filter != null) {
+            if ($filter == $filterLetter) {
+                $href .= 'filter_by_first_letter=' . $filter . '&';
+            } elseif ($filter == $filterState) {
+                $href .= 'filter_by_state=' . $filter . '&';
+            } elseif ($filter == $filterSort) {
+                $href .= 'sort=' . $filter;
+            }
+        }
+    }
+
+    if ($pageAmong < $filterPage + PAGINATING_INDEX ) {
+        $end = $pageAmong;
+    } else {
+        $end = $filterPage + PAGINATING_INDEX;
+    }
+    if ($filterPage > PAGINATING_INDEX) {
+        for ($i = $filterPage - PAGINATING_INDEX; $i < $end; $i++) {
+            if ($i == $filterPage) {
+                echo "<li class=\"page-item active\"><a class=\"page-link\" href=\"/?page=$i&$href\">$i</a></li>";
```

Comment:
Ok, it's possible, but it will be better to return some value and then use this value somewhere.

Reply:
Your right, changed to returning an array of html pieces in last commit